src/regex_manager.rs

```diff
@@ -90,7 +90,7 @@ impl RegexManager {
         if !filter.is_regex() && !filter.is_complete_regex() {
             return true;
         }
-        let key = filter as *const NetworkFilter;
```

Comment:
Im fact we inside the lock (for the multiple threads case). Frankly I'm don't fully understand what's wrong with taking the pointer.